FlowCrypt/src/main/java/com/flowcrypt/email/api/email/sync/tasks/DeleteMessagesPermanentlySyncTask.kt

```diff
@@ -0,0 +1,65 @@
+/*
+ * © 2016-present FlowCrypt a.s. Limitations apply. Contact human@flowcrypt.com
+ * Contributors: DenBond7
+ */
+
+package com.flowcrypt.email.api.email.sync.tasks
+
+import com.flowcrypt.email.api.email.FoldersManager
+import com.flowcrypt.email.api.email.sync.SyncListener
+import com.flowcrypt.email.database.FlowCryptRoomDatabase
+import com.flowcrypt.email.database.MessageState
+import com.flowcrypt.email.database.entity.AccountEntity
+import com.sun.mail.imap.IMAPFolder
+import javax.mail.Flags
+import javax.mail.Folder
+import javax.mail.Message
+import javax.mail.Session
+import javax.mail.Store
+
+/**
+ * @author Denis Bondarenko
+ *         Date: 7/3/20
+ *         Time: 5:33 PM
+ *         E-mail: DenBond7@gmail.com
+ */
+class DeleteMessagesPermanentlySyncTask(ownerKey: String,
+                                        requestCode: Int) : BaseSyncTask(ownerKey, requestCode) {
+
+  override fun runIMAPAction(account: AccountEntity, session: Session, store: Store, listener: SyncListener) {
+    val context = listener.context
+    val foldersManager = FoldersManager.fromDatabase(context, account.email)
+    val trash = foldersManager.folderTrash
+
+    if (trash == null) {
+      listener.onActionCompleted(account, ownerKey, requestCode)
+      return
+    }
+
+    val roomDatabase = FlowCryptRoomDatabase.getDatabase(context)
+
+    while (true) {
```

Comment:
There is some possible chance here for infinite loop - if the remote server didn't actually move or change items as expected. We could consider instead testing either `isEmpty` OR that the amount of items did not change from last cycle.

But if you think there is no chance, then no need.